File: src/LinearSearch/basic/LinearSearch.py
class LinearSearch:
    @staticmethod
    def search(data, target):
        for index, one in enumerate(data):
            if one == target:
                return index
        return -1


if __name__ == '__main__':
    test_list = [24, 18, 12, 9, 16, 66, 32, 4]
    print(LinearSearch.search(test_list, 16))
    print(LinearSearch.search(test_list, 666))

File: src/LinearSearch/basic/LinearSearch.ts
class LinearSearch {
    static search = function (arr: number[], target: number): number {
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] === target) {
                return i;
            }
        }
        return -1;
    }
}


const testList: number[] = [24, 18, 12, 9, 16, 66, 32, 4];
console.log(LinearSearch.search(testList, 16))
console.log(LinearSearch.search(testList, 666))